dbt/adapters/athena/column.py

```diff
@@ -39,6 +43,23 @@ def binary_type(cls) -> str:
     def timestamp_type(self) -> str:
         return "timestamp(6)" if self.is_iceberg() else "timestamp"
 
+    @classmethod
+    def array_type(cls, inner_type: str) -> str:
+        return f"array({inner_type})"
+
+    def array_inner_type(self) -> str:
+        if not self.is_array():
+            raise DbtRuntimeError("Called array_inner_type() on non-array field!")
```

Comment:
This error matches the pattern set by `AthenaColumn.string_size()`:

https://github.com/dbt-athena/dbt-athena/blob/7d850524d3a4f80635dcdeb71684bf8ef0a03b16/dbt/adapters/athena/column.py#L43-L44